Java 17 module access violation

Hello together!

### Bug description
we recently upgraded our application to Java 17 and after some time we recognized that the access with e.g. UaExpert/Prosys was broken. When trying to read the value of a Node with a custom structure, I can't read the value. That worked in the past. 

### Logs
When I pack my application into a jar and run it, I get this exception:
```Apr 14, 2022 7:58:07 AM com.sun.xml.bind.v2.runtime.reflect.opt.Injector <clinit>
SEVERE: null
java.security.PrivilegedActionException: java.lang.NoSuchMethodException: sun.misc.Unsafe.defineClass(java.lang.String,[B,int,int,java.lang.ClassLoader,java.security.ProtectionDomain)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:573)
        at com.sun.xml.bind.v2.runtime.reflect.opt.Injector.<clinit>(Injector.java:166)
        at com.sun.xml.bind.v2.runtime.reflect.opt.AccessorInjector.prepare(AccessorInjector.java:51)
        at com.sun.xml.bind.v2.runtime.reflect.opt.OptimizedAccessorFactory.get(OptimizedAccessorFactory.java:157)
        at com.sun.xml.bind.v2.runtime.reflect.Accessor$FieldReflection.optimize(Accessor.java:255)
        at com.sun.xml.bind.v2.runtime.property.SingleElementNodeProperty.<init>(SingleElementNodeProperty.java:64)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at com.sun.xml.bind.v2.runtime.property.PropertyFactory.create(PropertyFactory.java:99)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.<init>(ClassBeanInfoImpl.java:150)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.getOrCreate(JAXBContextImpl.java:493)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.<init>(ClassBeanInfoImpl.java:137)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.getOrCreate(JAXBContextImpl.java:493)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.<init>(ClassBeanInfoImpl.java:137)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.getOrCreate(JAXBContextImpl.java:493)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.<init>(JAXBContextImpl.java:310)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.<init>(JAXBContextImpl.java:109)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl$JAXBContextBuilder.build(JAXBContextImpl.java:1142)
        at com.sun.xml.bind.v2.ContextFactory.createContext(ContextFactory.java:141)
        at com.sun.xml.bind.v2.JAXBContextFactory.createContext(JAXBContextFactory.java:35)
        at javax.xml.bind.ContextFinder.find(ContextFinder.java:358)
        at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:691)
        at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:632)
        at org.eclipse.milo.opcua.binaryschema.generator.DataTypeDictionaryGenerator.writeToOutputStream(DataTypeDictionaryGenerator.java:108)
        at org.eclipse.milo.opcua.sdk.server.dtd.DataTypeDictionaryManager.writeDictionaryToStream(DataTypeDictionaryManager.java:403)
        at org.eclipse.milo.opcua.sdk.server.dtd.DataTypeDictionaryManager.writeDictionaryToFile(DataTypeDictionaryManager.java:365)
        at org.eclipse.milo.opcua.sdk.server.dtd.DataTypeDictionaryManager.lambda$null$0(DataTypeDictionaryManager.java:129)
        at org.eclipse.milo.opcua.stack.core.util.Lazy.maybeCompute(Lazy.java:50)
        at org.eclipse.milo.opcua.stack.core.util.Lazy.getOrCompute(Lazy.java:36)
        at org.eclipse.milo.opcua.sdk.server.dtd.DataTypeDictionaryManager.lambda$startup$1(DataTypeDictionaryManager.java:127)
        at org.eclipse.milo.opcua.sdk.server.nodes.filters.AttributeFilters$1.getAttribute(AttributeFilters.java:30)
        at org.eclipse.milo.opcua.sdk.server.nodes.filters.AttributeFilterChain.getAttribute(AttributeFilterChain.java:82)
        at org.eclipse.milo.opcua.sdk.server.nodes.UaNode.getAttribute(UaNode.java:647)
        at org.eclipse.milo.opcua.sdk.server.util.AttributeReader.readAttribute(AttributeReader.java:96)
        at org.eclipse.milo.opcua.sdk.server.nodes.UaServerNode.readAttribute(UaServerNode.java:114)
        at org.eclipse.milo.opcua.sdk.server.nodes.UaServerNode.lambda$readAttribute$0(UaServerNode.java:91)
        at java.base/java.util.Optional.map(Optional.java:260)
        at org.eclipse.milo.opcua.sdk.server.nodes.UaServerNode.readAttribute(UaServerNode.java:91)
        at org.eclipse.milo.opcua.sdk.server.api.ManagedAddressSpace.read(ManagedAddressSpace.java:149)
        at de.cloos.cgate.opcua.adapter.OpcUaObserverNamespace.read(OpcUaObserverNamespace.java:322)
        at org.eclipse.milo.opcua.sdk.server.api.AddressSpaceComposite.lambda$null$16(AddressSpaceComposite.java:330)
        at org.eclipse.milo.opcua.sdk.core.util.GroupMapCollate.lambda$groupMapCollate$3(GroupMapCollate.java:47)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1850)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
        at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
        at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
        at org.eclipse.milo.opcua.sdk.core.util.GroupMapCollate.allOf(GroupMapCollate.java:77)
        at org.eclipse.milo.opcua.sdk.core.util.GroupMapCollate.groupMapCollate(GroupMapCollate.java:68)
        at org.eclipse.milo.opcua.sdk.server.api.AddressSpaceComposite.read(AddressSpaceComposite.java:317)
        at org.eclipse.milo.opcua.sdk.server.services.DefaultAttributeServiceSet.onRead(DefaultAttributeServiceSet.java:71)
        at org.eclipse.milo.opcua.sdk.server.SessionManager.onRead(SessionManager.java:871)
        at org.eclipse.milo.opcua.stack.server.UaStackServer.handleServiceRequest(UaStackServer.java:313)
        at org.eclipse.milo.opcua.stack.server.UaStackServer.lambda$onServiceRequest$10(UaStackServer.java:276)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NoSuchMethodException: sun.misc.Unsafe.defineClass(java.lang.String,[B,int,int,java.lang.ClassLoader,java.security.ProtectionDomain)
        at java.base/java.lang.Class.getMethod(Class.java:2227)
        at com.sun.xml.bind.v2.runtime.reflect.opt.Injector$3.run(Injector.java:170)
        at com.sun.xml.bind.v2.runtime.reflect.opt.Injector$3.run(Injector.java:166)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
        ... 61 more

WARN  [milo-shared-thread-pool-2] [2022-04-14 07:58:07,930] [org.eclipse.milo.opcua.sdk.server.dtd.DataTypeDictionaryManager]  -  Failed to write dictionary file
java.lang.RuntimeException: failed to write dictionary file
        at org.eclipse.milo.opcua.sdk.server.dtd.DataTypeDictionaryManager.lambda$null$0(DataTypeDictionaryManager.java:131)
        at org.eclipse.milo.opcua.stack.core.util.Lazy.maybeCompute(Lazy.java:50)
        at org.eclipse.milo.opcua.stack.core.util.Lazy.getOrCompute(Lazy.java:36)
        at org.eclipse.milo.opcua.sdk.server.dtd.DataTypeDictionaryManager.lambda$startup$1(DataTypeDictionaryManager.java:127)
        at org.eclipse.milo.opcua.sdk.server.nodes.filters.AttributeFilters$1.getAttribute(AttributeFilters.java:30)
        at org.eclipse.milo.opcua.sdk.server.nodes.filters.AttributeFilterChain.getAttribute(AttributeFilterChain.java:82)
        at org.eclipse.milo.opcua.sdk.server.nodes.UaNode.getAttribute(UaNode.java:647)
        at org.eclipse.milo.opcua.sdk.server.util.AttributeReader.readAttribute(AttributeReader.java:96)
        at org.eclipse.milo.opcua.sdk.server.nodes.UaServerNode.readAttribute(UaServerNode.java:114)
        at org.eclipse.milo.opcua.sdk.server.nodes.UaServerNode.lambda$readAttribute$0(UaServerNode.java:91)
        at java.base/java.util.Optional.map(Optional.java:260)
        at org.eclipse.milo.opcua.sdk.server.nodes.UaServerNode.readAttribute(UaServerNode.java:91)
        at org.eclipse.milo.opcua.sdk.server.api.ManagedAddressSpace.read(ManagedAddressSpace.java:149)
        at de.cloos.cgate.opcua.adapter.OpcUaObserverNamespace.read(OpcUaObserverNamespace.java:322)
        at org.eclipse.milo.opcua.sdk.server.api.AddressSpaceComposite.lambda$null$16(AddressSpaceComposite.java:330)
        at org.eclipse.milo.opcua.sdk.core.util.GroupMapCollate.lambda$groupMapCollate$3(GroupMapCollate.java:47)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1850)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
        at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
        at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
        at org.eclipse.milo.opcua.sdk.core.util.GroupMapCollate.allOf(GroupMapCollate.java:77)
        at org.eclipse.milo.opcua.sdk.core.util.GroupMapCollate.groupMapCollate(GroupMapCollate.java:68)
        at org.eclipse.milo.opcua.sdk.server.api.AddressSpaceComposite.read(AddressSpaceComposite.java:317)
        at org.eclipse.milo.opcua.sdk.server.services.DefaultAttributeServiceSet.onRead(DefaultAttributeServiceSet.java:71)
        at org.eclipse.milo.opcua.sdk.server.SessionManager.onRead(SessionManager.java:871)
        at org.eclipse.milo.opcua.stack.server.UaStackServer.handleServiceRequest(UaStackServer.java:313)
        at org.eclipse.milo.opcua.stack.server.UaStackServer.lambda$onServiceRequest$10(UaStackServer.java:276)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.io.IOException: failed to write dictionary to OutputStream
        at org.eclipse.milo.opcua.binaryschema.generator.DataTypeDictionaryGenerator.writeToOutputStream(DataTypeDictionaryGenerator.java:123)
        at org.eclipse.milo.opcua.sdk.server.dtd.DataTypeDictionaryManager.writeDictionaryToStream(DataTypeDictionaryManager.java:403)
        at org.eclipse.milo.opcua.sdk.server.dtd.DataTypeDictionaryManager.writeDictionaryToFile(DataTypeDictionaryManager.java:365)
        at org.eclipse.milo.opcua.sdk.server.dtd.DataTypeDictionaryManager.lambda$null$0(DataTypeDictionaryManager.java:129)
        ... 32 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.reflect.Method.invoke(Object, Object[])" because "com.sun.xml.bind.v2.runtime.reflect.opt.Injector.defineClass" is null
        at com.sun.xml.bind.v2.runtime.reflect.opt.Injector.inject(Injector.java:294)
        at com.sun.xml.bind.v2.runtime.reflect.opt.Injector.inject(Injector.java:66)
        at com.sun.xml.bind.v2.runtime.reflect.opt.AccessorInjector.prepare(AccessorInjector.java:57)
        at com.sun.xml.bind.v2.runtime.reflect.opt.OptimizedAccessorFactory.get(OptimizedAccessorFactory.java:157)
        at com.sun.xml.bind.v2.runtime.reflect.Accessor$FieldReflection.optimize(Accessor.java:255)
        at com.sun.xml.bind.v2.runtime.property.SingleElementNodeProperty.<init>(SingleElementNodeProperty.java:64)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at com.sun.xml.bind.v2.runtime.property.PropertyFactory.create(PropertyFactory.java:99)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.<init>(ClassBeanInfoImpl.java:150)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.getOrCreate(JAXBContextImpl.java:493)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.<init>(ClassBeanInfoImpl.java:137)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.getOrCreate(JAXBContextImpl.java:493)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.<init>(ClassBeanInfoImpl.java:137)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.getOrCreate(JAXBContextImpl.java:493)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.<init>(JAXBContextImpl.java:310)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.<init>(JAXBContextImpl.java:109)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl$JAXBContextBuilder.build(JAXBContextImpl.java:1142)
        at com.sun.xml.bind.v2.ContextFactory.createContext(ContextFactory.java:141)
        at com.sun.xml.bind.v2.JAXBContextFactory.createContext(JAXBContextFactory.java:35)
        at javax.xml.bind.ContextFinder.find(ContextFinder.java:358)
        at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:691)
        at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:632)
        at org.eclipse.milo.opcua.binaryschema.generator.DataTypeDictionaryGenerator.writeToOutputStream(DataTypeDictionaryGenerator.java:108)
        ... 35 more
```
### Solution
I managed to solve this, by forcing gradle to use version 2.3.6 of org.glassfish.jaxb:jaxb-runtime. Could you please upgrade the dependencies as well?

### Context
Java 17.0.2 (OpenJDK), Milo 0.6.4

--- a/opc-ua-stack/bsd-core/pom.xml
+++ b/opc-ua-stack/bsd-core/pom.xml
@@ -24,13 +24,13 @@
         <dependency>
             <groupId>com.sun.activation</groupId>
             <artifactId>jakarta.activation</artifactId>
-            <version>1.2.1</version>
+            <version>1.2.2</version>
         </dependency>
 
         <dependency>
             <groupId>org.glassfish.jaxb</groupId>
             <artifactId>jaxb-runtime</artifactId>
-            <version>2.3.2</version>
+            <version>2.3.6</version>
         </dependency>
     </dependencies>
 

--- a/opc-ua-stack/bsd-parser/pom.xml
+++ b/opc-ua-stack/bsd-parser/pom.xml
@@ -24,13 +24,13 @@
         <dependency>
             <groupId>com.sun.activation</groupId>
             <artifactId>jakarta.activation</artifactId>
-            <version>1.2.1</version>
+            <version>1.2.2</version>
         </dependency>
 
         <dependency>
             <groupId>org.glassfish.jaxb</groupId>
             <artifactId>jaxb-runtime</artifactId>
-            <version>2.3.3</version>
+            <version>2.3.6</version>
         </dependency>
 
         <dependency>

--- a/opc-ua-stack/stack-core/pom.xml
+++ b/opc-ua-stack/stack-core/pom.xml
@@ -74,7 +74,7 @@
         <dependency>
             <groupId>org.glassfish.jaxb</groupId>
             <artifactId>jaxb-runtime</artifactId>
-            <version>2.3.3</version>
+            <version>2.3.6</version>
         </dependency>
 
         <dependency>

